src/types.ts

```diff
@@ -67,6 +67,16 @@ export interface PaymentNotificationData {
   };
 }
 
+export interface AuthResponseObject {
+  reason?: string;
+  status: string;
+}
+export interface AuthNotificationData {
+  authResponse?: AuthResponseObject;
+  origin?: OriginData;
```

Comment:
So many optional values, a comment would be good like

Is only defined if ...
Is always undefined if ...